lookout/style/format/tests/test_features.py

```diff
@@ -116,6 +117,14 @@ def test_noop_vnodes(self):
             else:
                 self.assertEqual(vnode.y, CLASS_INDEX[CLS_NOOP])
 
+    def test_extended_roles(self):
+        file = File(content=bytes(self.contents, 'utf-8'),
+                    uast=self.uast)
+        X, _, _ = self.extractor.extract_features([file])
+        # last columns are only roles
+        last_columns = self.extractor.parents_depth + self.extractor.siblings_window
+        self.assertGreater(numpy.count_nonzero(X[:, -last_columns:] > len(ROLE_INDEX)), 0)
```

Comment:
We also return vnodes corresponding to X. Can you please add the check for vnode.value in RESERVED_INDEX

Reply:
Actually we only have roles for the neighbours of the vnode of `X` (the role of `X` itself being the target), so it might be hard and `vnode.value` might not be in `RESERVED_INDEX` for legitimate reasons.